File: src/henaxel/workbench/Tool.java
package henaxel.workbench;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;

public class Tool extends JPanel {

    public Tool() {

    }
    public Tool(LayoutManager layoutMGR) {
        this.setLayout(layoutMGR);
    }

    
    public static Tool buildButtonTool(String label, String toolName, int iconSize, AbstractAction abstractAction) {
        JButton button = new JButton();
    
        if(button.getClass().getResource(String.format("res/%s%d.png", toolName, iconSize)) != null) {
            button.setIcon(new ImageIcon(button.getClass().getResource(String.format("res/saveImage%d.png", iconSize))));
        } else {
            button.setText(label);
            button.setFont(new Font("Dialog", Font.PLAIN, 10));
        }
    
        button.setBackground(Color.white);
        button.setBorder(BorderFactory.createLineBorder(Color.lightGray, 1));
        button.addActionListener(abstractAction);
    
        Tool tool = new Tool();
        tool.setLayout(new BorderLayout());
        tool.add(button, BorderLayout.CENTER);
        return tool;
    }
    
    public static Tool buildToggleButtonTool(String label, String toolName, int iconSize, boolean selected, AbstractAction abstractAction) {
        JToggleButton button = new JToggleButton();
        button.setSelected(selected);
        if(button.getClass().getResource(String.format("res/%s%d.png", toolName, iconSize)) != null) {
            button.setIcon(new ImageIcon(button.getClass().getResource(String.format("res/saveImage%d.png", iconSize))));
        } else {
            button.setText(label);
            button.setFont(new Font("Dialog", Font.PLAIN, 10));
        }
        
        button.setBackground(Color.white);
        button.setBorder(BorderFactory.createLineBorder(Color.lightGray, 1));
        button.addActionListener(abstractAction);
        
        Tool tool = new Tool();
        tool.setLayout(new BorderLayout());
        tool.add(button, BorderLayout.CENTER);
        return tool;
    }
}

File: src/henaxel/workbench/Tool.kt
package henaxel.workbench

import java.awt.BorderLayout
import java.awt.Color
import java.awt.Font
import java.awt.LayoutManager
import javax.swing.*

class Tool() : JPanel() {

    constructor(layoutManager: LayoutManager) : this() {
        layout = layoutManager
    }

    companion object {

        fun buildButtonTool(label: String, name: String, iconSize: Int, action: AbstractAction): Tool {
            val button = JButton().apply {

                if (javaClass.getResource("res/${name}${iconSize}.png") != null) {
                    icon = ImageIcon(this.javaClass.getResource("res/${name}${iconSize}.png"))
                } else {
                    text = label
                    font = Font("Dialog", Font.PLAIN, 10)
                }

                background = Color.white
                border = BorderFactory.createLineBorder(Color.lightGray, 1)
                addActionListener(action)
            }

            return Tool().apply {
                layout = BorderLayout()
                add(button, BorderLayout.CENTER)
            }
        }

        fun buildToggleButtonTool(label: String, name: String, iconSize: Int, selected: Boolean, action: AbstractAction): Tool {
            val button = JToggleButton().apply {
                isSelected = selected
                if (javaClass.getResource("res/${name}${iconSize}.png") != null) {
                    icon = ImageIcon(this.javaClass.getResource("res/${name}${iconSize}.png"))
                } else {
                    text = label
                    font = Font("Dialog", Font.PLAIN, 10)
                }

                background = Color.white
                border = BorderFactory.createLineBorder(Color.lightGray, 1)
                addActionListener(action)
            }

            return Tool().apply {
                layout = BorderLayout()
                add(button, BorderLayout.CENTER)
            }
        }
    }
}